app/assets/javascripts/comfortable_mexican_sofa/admin/modules/ArticleComponents.js

```diff
@@ -0,0 +1,81 @@
+define(['jquery','DoughBaseComponent'], function ($, DoughBaseComponent) {
+  'use strict';
+
+  var ArticleComponentsProto,
+      defaultConfig = {};
+
+  function ArticleComponents($el, config) {
+    ArticleComponents.baseConstructor.call(this, $el, config, defaultConfig);
+
+    this.$articleComponents = $el.find('[data-dough-article-component]');
+    this.activeClass = 'is-active';
+  }
+
+  DoughBaseComponent.extend(ArticleComponents);
+
+  ArticleComponents.componentName = 'ArticleComponents';
+
+  ArticleComponentsProto = ArticleComponents.prototype;
+
+  ArticleComponentsProto.init = function(initialised) {
+    this._initialisedSuccess(initialised);
+    this._setUpComponent();
+    this._checkInitialState();
+
+    return this;
+  };
+
+  ArticleComponentsProto._setUpComponent = function() {
+    var self = this,
+        component_add =
+          '<button class="article-component--add button button--secondary button--action">' +
```

Comment:
Markup preferably shouldn't be in JS as it's hidden away. There isn't a great solution to this to be honest. Could we potentially have the divs on the page on load and hide/show them?

Reply:
@rhianwatkins1 I agree on this but these elements are generated via the Form Builder which makes it hard to customise. Also, the CMS already has a number of instances of this so I have slightly reluctantly continued the trend here.

Reply:
Fine by me!